worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/filter/LinkedFilter.java

```diff
@@ -1,69 +1,102 @@
 package com.fastasyncworldedit.core.extent.filter;
 
-import com.fastasyncworldedit.core.extent.filter.block.DelegateFilter;
 import com.fastasyncworldedit.core.extent.filter.block.FilterBlock;
 import com.fastasyncworldedit.core.internal.simd.VectorizedFilter;
 import com.fastasyncworldedit.core.queue.Filter;
+import com.fastasyncworldedit.core.queue.IChunk;
+import com.sk89q.worldedit.regions.Region;
 import jdk.incubator.vector.ShortVector;
+import org.jetbrains.annotations.Nullable;
 
 /**
- * Filter which links two Filters together for single-filter-input operations.
+ * Filter which links two Filters together for single-filter-input operations. Left filter is operated first.
  *
- * @param <T> Parent which extends Filter
- * @param <S> Child which extends Filter
+ * @param <L> Left filter
+ * @param <R> Right filter
  */
-public sealed class LinkedFilter<T extends Filter, S extends Filter> extends DelegateFilter<T> {
+public sealed class LinkedFilter<L extends Filter, R extends Filter> implements Filter {
 
-    private final S child;
+    private final L left;
+    private final R right;
+
+    public LinkedFilter(L left, R right) {
+        this.left = left;
+        this.right = right;
+    }
 
     @SuppressWarnings({"unchecked", "rawtypes"}) // we defeated the type system
-    public static <T extends Filter, S extends Filter> LinkedFilter<? extends T, ? extends S> of(T parent, S child) {
-        if (parent instanceof VectorizedFilter p && child instanceof VectorizedFilter c) {
-            return new VectorizedLinkedFilter(p, c);
+    public static <L extends Filter, R extends Filter> LinkedFilter<? extends L, ? extends R> of(L left, R right) {
+        if (left instanceof VectorizedFilter l && right instanceof VectorizedFilter r) {
+            return new VectorizedLinkedFilter(l, r);
         }
-        return new LinkedFilter<>(parent, child);
+        return new LinkedFilter<>(left, right);
+    }
+
+    public L getLeft() {
+        return this.left;
+    }
+
+    public R getRight() {
+        return this.right;
+    }
+
+    @Override
+    public boolean appliesChunk(int chunkX, int chunkZ) {
+        return getLeft().appliesChunk(chunkX, chunkZ) && getRight().appliesChunk(chunkX, chunkZ);
     }
 
-    public LinkedFilter(T parent, S child) {
-        super(parent);
-        this.child = child;
+    @Override
+    public <T extends IChunk> T applyChunk(T chunk, @Nullable Region region) {
+        chunk = getLeft().applyChunk(chunk, region);
+        return getRight().applyChunk(chunk, region);
     }
 
-    public S getChild() {
-        return this.child;
+    @Override
+    public boolean appliesLayer(IChunk chunk, int layer) {
+        return getLeft().appliesLayer(chunk, layer) && getRight().appliesLayer(chunk, layer);
```

Comment:
This should be `||`, shouldn't it?

Reply:
I wasn't really sure what was best. Do we want to still filter in one if the other doesn't want to? That implies that edit shouldn't be happening in the layer

Reply:
My understanding was that the methods basically mean return whether a filter *may* do something with the chunk/layer. Now if we say "right only needs to do something if left did something", then AND should be fine, but otherwise not. There's also a theoretical scenario where right can only decide whether it needs to do something *after* left already did its work. We currently don't have any custom implementations for those methods (the default returns true, other impls just delegate), so maybe it would make sense to just remove them? And if we ever think that something like that might be useful, we can reintroduce it and think about it again, having an actual example at hand.

Reply:
Mm that might be the best solution for now, setting the return for applyChunk to nonnull I suppose